### ISSUE TITLE
Could not read typeID: read only 1 of 4 bytes from solverl/lib/sasl-3.4/ebin/erlsrv.beam

### ISSUE THREAD

**Issue by bokner** [KronicDeth/intellij-elixir#1816]
# Exception

## Stacktrace

```
java.lang.Throwable: Could not read typeID: read only 1 of 4 bytes from <project_dir>/_build/dev/rel/solverl/lib/sasl-3.4/ebin/erlsrv.beam
	at com.intellij.openapi.diagnostic.Logger.error(Logger.java:134)
	at org.elixir_lang.beam.chunk.Chunk.typeID(Chunk.java:75)
	at org.elixir_lang.beam.chunk.Chunk.from(Chunk.java:36)
	at org.elixir_lang.beam.Beam$Companion.from(Beam.kt:152)
	at org.elixir_lang.beam.Beam$Companion.from(Beam.kt:176)
	at org.elixir_lang.beam.psi.BeamFileImpl.buildFileStub(BeamFileImpl.java:90)
	at org.elixir_lang.beam.StubBuilder.buildStubTree(StubBuilder.java:31)
	at com.intellij.psi.stubs.StubTreeBuilder.buildStubTree(StubTreeBuilder.java:61)
	at com.intellij.psi.stubs.StubUpdatingIndex$3.lambda$map$0(StubUpdatingIndex.java:238)
	at com.intellij.openapi.application.impl.ApplicationImpl.runReadAction(ApplicationImpl.java:943)
	at com.intellij.psi.stubs.StubUpdatingIndex$3.map(StubUpdatingIndex.java:220)
	at com.intellij.psi.stubs.StubUpdatingIndex$3.map(StubUpdatingIndex.java:202)
	at com.intellij.util.indexing.impl.MapReduceIndex.mapInput(MapReduceIndex.java:276)
	at com.intellij.util.indexing.VfsAwareMapReduceIndex.calculateUpdateData(VfsAwareMapReduceIndex.java:83)
	at com.intellij.util.indexing.impl.MapReduceIndex.update(MapReduceIndex.java:216)
	at com.intellij.util.indexing.FileBasedIndexImpl.updateSingleIndex(FileBasedIndexImpl.java:1668)
	at com.intellij.util.indexing.FileBasedIndexImpl.lambda$doIndexFileContent$27(FileBasedIndexImpl.java:1596)
	at com.intellij.openapi.fileTypes.impl.FileTypeManagerImpl.freezeFileTypeTemporarilyIn(FileTypeManagerImpl.java:522)
	at com.intellij.util.indexing.FileBasedIndexImpl.doIndexFileContent(FileBasedIndexImpl.java:1555)
	at com.intellij.util.indexing.FileBasedIndexImpl.indexFileContent(FileBasedIndexImpl.java:1543)
	at com.intellij.util.indexing.UnindexedFilesUpdater.lambda$indexFiles$0(UnindexedFilesUpdater.java:110)
	at com.intellij.openapi.project.CacheUpdateRunner$MyRunnable.lambda$run$0(CacheUpdateRunner.java:230)
	at com.intellij.openapi.application.impl.ApplicationImpl.tryRunReadAction(ApplicationImpl.java:1168)
	at com.intellij.openapi.project.CacheUpdateRunner$MyRunnable.lambda$run$1(CacheUpdateRunner.java:246)
	at com.intellij.openapi.progress.impl.CoreProgressManager.lambda$runProcess$2(CoreProgressManager.java:164)
	at com.intellij.openapi.progress.impl.CoreProgressManager.registerIndicatorAndRun(CoreProgressManager.java:582)
	at com.intellij.openapi.progress.impl.CoreProgressManager.executeProcessUnderProgress(CoreProgressManager.java:532)
	at com.intellij.openapi.progress.impl.ProgressManagerImpl.executeProcessUnderProgress(ProgressManagerImpl.java:87)
	at com.intellij.openapi.progress.impl.CoreProgressManager.runProcess(CoreProgressManager.java:151)
	at com.intellij.openapi.project.CacheUpdateRunner$MyRunnable.run(CacheUpdateRunner.java:243)
	at com.intellij.openapi.application.impl.ApplicationImpl$1.run(ApplicationImpl.java:314)
	at java.util.concurrent.Executors$RunnableAdapter.call(Executors.java:511)
	at java.util.concurrent.FutureTask.run(FutureTask.java:266)
	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1142)
	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:617)
	at java.lang.Thread.run(Thread.java:745)
```

**Comment by sh41** [comment]
Fixed in v11.1.0, September 2019.

The message means the file held fewer than four bytes when it was read — it says nothing about the
module in the path. A real `erlsrv.beam` decompiles fine, and I could only reproduce the message by
truncating a `.beam` to one, two or three bytes. A release's `lib/<app>-<version>/ebin` is filled while
the IDE is indexing, so a read landing mid-copy fits.

The log was removed deliberately, in 56b7c0900 and 88cc56783: a partially-written file isn't
actionable, and the plugin picks it up once it is complete.

If a `.beam` still fails to open for you, that is a different bug and worth its own issue.

Closed as part of an AI-assisted review of the backlog. A human checked this one before closing it, but
it might still be a mistake. If you think the reason above is wrong, please reopen the issue and tell us
why.
